src/core/ext/filters/client_channel/http_connect_handshaker.cc

```diff
@@ -254,7 +254,8 @@ static void http_connect_handshaker_do_handshake(
   // If not found, invoke on_handshake_done without doing anything.
   const grpc_arg* arg =
       grpc_channel_args_find(args->args, GRPC_ARG_HTTP_CONNECT_SERVER);
-  if (arg == nullptr) {
+  if (arg == nullptr || arg->type != GRPC_ARG_STRING) {
+    gpr_log(GPR_INFO, "HTTP CONNECT channel arg not found or invalid");
```

Comment:
This log message is now going to show up for every single subchannel connection attempt, even if HTTP CONNECT is not configured.  That's probably a little too verbose.

Also, you can remove the GPR_ASSERT on line 267 below, since it will now never be reached.

Reply:
Done: #14427